pkg/apis/workflow/v1alpha1/types.go

```diff
@@ -135,6 +135,18 @@ type WorkflowSpec struct {
 	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
 	// Priority is used if controller is configured to process limited number of workflows in parallel. Workflows with higher priority are processed first.
 	Priority *int32 `json:"priority,omitempty"`
+
+	// Restart policy for all containers within all the pods.
+	// One of Always, OnFailure, Never.
+	// Default to Always.
+	// More info: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy
+	// +optional
+	RestartPolicy apiv1.RestartPolicy `json:"restartPolicy,omitempty"`
```

Comment:
I think we should be consistent with Job and allow only `Never` or `OnFailure` values.